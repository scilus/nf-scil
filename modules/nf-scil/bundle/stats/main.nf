process BUNDLE_STATS {
    tag "$meta.id"
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://scil.usherbrooke.ca/containers/scilus_2.0.2.sif':
        'scilus/scilus:2.0.2' }"

    input:
    tuple val(meta), path(bundles), path(labels_map), path(metrics), path(lesions)

    output:
    tuple val(meta), path("*_length_stats.json")                , emit: length, optional: true
    tuple val(meta), path("*_endpoints_map_raw.json")           , emit: endpoints_raw, optional: true
    tuple val(meta), path("*_endpoints_metric_stats.json")      , emit: endpoints_metric_stats, optional: true
    tuple val(meta), path("*_mean_std.json")                    , emit: mean_std, optional: true
    tuple val(meta), path("*_volume.json")                      , emit: volume, optional: true
    tuple val(meta), path("*_volume_lesions.json")              , emit: volume_lesions, optional: true
    tuple val(meta), path("*_streamline_count.json")            , emit: streamline_count, optional: true
    tuple val(meta), path("*_streamline_count_lesions.json")    , emit: streamline_count_lesions, optional: true
    tuple val(meta), path("*_volume_per_label.json")            , emit: volume_per_labels, optional: true
    tuple val(meta), path("*_volume_per_label_lesions.json")    , emit: volume_per_labels_lesions, optional: true
    tuple val(meta), path("*_mean_std_per_point.json")          , emit: mean_std_per_point, optional: true
    tuple val(meta), path("*__lesion_stats.json")               , emit: lesion_stats, optional: true
    tuple val(meta), path("*_endpoints_map_head.nii.gz")        , emit: endpoints_head, optional: true
    tuple val(meta), path("*_endpoints_map_tail.nii.gz")        , emit: endpoints_tail, optional: true
    tuple val(meta), path("*_lesion_map.nii.gz")                , emit: lesion_map, optional: true
    path "versions.yml"                                         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def density_weighting = task.ext.density_weighting ? "--density_weighting" : ""
    def normalize_weights = task.ext.normalize_weights ? "--normalize_weights" : "--bin"
    def length_stats = task.ext.length_stats ?: ""
    def endpoints = task.ext.endpoints ?: ""
    def mean_std = task.ext.mean_std ?: ""
    def volume = task.ext.volume ?: ""
    def lesions_stats = task.ext.lesions_stats ?: ""
    def min_lesion_vol = task.ext.min_lesion_vol ?: ""
    def streamline_count = task.ext.streamline_count ?: ""
    def volume_per_labels = task.ext.volume_per_labels ?: ""
    def mean_std_per_point = task.ext.mean_std_per_point ?: ""

    """
    bundles=( ${bundles.join(" ")} )
    label_map=( ${labels_map.join(" ")} )

    for index in \${!bundles[@]};
    do\
    bname=\$(basename \${bundles[index]} .trk);
    b_metrics="$metrics";

    if [[ "$length_stats" ]];
    then
        scil_tractogram_print_info.py \${bundles[index]} > \${bname}_length.json
    fi

    if [[ "$endpoints" ]];
    then
        scil_bundle_compute_endpoints_map.py \${bundles[index]} \
            \${bname}_endpoints_map_head.nii.gz \
            \${bname}_endpoints_map_tail.nii.gz >\
            ${prefix}__\${bname}_endpoints_raw.json;

        scil_volume_stats_in_ROI.py \${bname}_endpoints_map_head.nii.gz $normalize_weights\
            --metrics \${b_metrics} > \${bname}_head.json
        scil_volume_stats_in_ROI.py \${bname}_endpoints_map_tail.nii.gz $normalize_weights\
            --metrics \${b_metrics} > \${bname}_tail.json;

    fi

    if [[ "$mean_std" ]];
    then
        scil_bundle_mean_std.py $density_weighting \${bundles[index]} \${b_metrics} >\
            \${bname}_std.json
    fi

    if [[ "$volume" ]];
    then
        scil_bundle_shape_measures.py \${bundles[index]} > \${bname}_volume_stat.json

        if [[ "$lesions_stats" ]];
        then
            scil_lesions_info.py $lesions \${bname}_volume_lesions_stat.json \
                --bundle \${bundles[index]} --out_lesion_stats ${prefix}__lesion_stats.json \
                --out_streamlines_stats \${bname}_streamline_count_lesions_stat.json \
                --min_lesion_vol $min_lesion_vol -f
            fi

    elif [[ "$streamline_count" ]];
    then
        scil_tractogram_count_streamlines.py \${bundles[index]} > \${bname}_streamlines.json
    fi

    if [[ "$volume_per_labels" ]];
    then
        scil_bundle_volume_per_label.py \${label_map[index]} \$bname --sort_keys >\
            \${bname}_volume_label.json

        if [[ "$lesions_stats" ]];
        then
            scil_analyse_lesions_load.py $lesions \${bname}_volume_per_label_lesions_stat.json \
                --bundle_labels_map \${label_map[index]} \
                --out_lesion_atlas "${prefix}__\${bundles[index]}_lesion_map.nii.gz" \
                --min_lesion_vol $min_lesion_vol
            fi
    fi

    if [[ "$mean_std_per_point" ]];
    then
        scil_bundle_mean_std.py \${bundles[index]} \${b_metrics}\
            --per_point \${label_map[index]} --sort_keys $density_weighting > \${bname}_std_per_point.json
    fi;

    done

    #Bundle_Length_Stats
    if [[ "$length_stats" ]];
    then
        scil_json_merge_entries.py *_length.json ${prefix}_length_stats.json --add_parent_key ${prefix} \
                --keep_separate
    fi

    #Bundle_Endpoints_Map
    if [[ "$endpoints" ]];
    then
        scil_json_merge_entries.py *_endpoints_raw.json ${prefix}_endpoints_map_raw.json \
            --no_list --add_parent_key ${prefix}

    #Bundle_Metrics_Stats_In_Endpoints

        scil_json_merge_entries.py *_tail.json *_head.json ${prefix}_endpoints_metric_stats.json \
            --no_list --add_parent_key ${prefix}
    fi

    #Bundle_Mean_Std
    if [[ "$mean_std" ]];
    then
        scil_json_merge_entries.py *_std.json ${prefix}_mean_std.json --no_list --add_parent_key ${prefix}
    fi

    #Bundle_Volume
    if [[ "$volume" ]];
    then
        scil_json_merge_entries.py *_volume_stat.json ${prefix}_volume.json --no_list --add_parent_key ${prefix}

        if [[ "$lesions_stats" ]];
        then
            scil_json_merge_entries.py *_volume_lesions_stat.json ${prefix}_volume_lesions.json --no_list --add_parent_key ${prefix}
            scil_json_merge_entries.py *_streamline_count_lesions_stat.json ${prefix}_streamline_count_lesions.json \
                --no_list --add_parent_key ${prefix}
            scil_merge_json.py ${prefix}__lesion_stats.json ${prefix}__lesion_stats.json \
                --remove_parent_key --add_parent_key ${prefix} -f
        fi

    #Bundle_Streamline_Count
    elif [[ "$streamline_count" ]];
    then
        scil_json_merge_entries.py *_streamlines.json ${prefix}_streamline_count.json --no_list \
            --add_parent_key ${prefix}
    fi

    #Bundle_Volume_Per_Label
    if [[ "$volume_per_labels" ]];
    then
        scil_json_merge_entries.py *_volume_label.json ${prefix}_volume_per_label.json --no_list \
            --add_parent_key ${prefix}

        if [[ "$lesions_stats" ]];
        then
        scil_json_merge_entries.py *_volume_per_label_lesions_stat.json ${prefix}_volume_per_label_lesions.json \
            --no_list --add_parent_key ${prefix}
        fi
    fi

    #Bundle_Mean_Std_Per_Point
    if [[ "$mean_std_per_point" ]];
    then
        scil_json_merge_entries.py *_std_per_point.json ${prefix}_mean_std_per_point.json --no_list \
            --add_parent_key ${prefix}
    fi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: 2.0.2
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    scil_tractogram_print_info.py -h
    scil_bundle_compute_endpoints_map.py -h
    scil_volume_stats_in_ROI.py -h
    scil_bundle_mean_std.py -h
    scil_bundle_shape_measures.py -h
    scil_tractogram_count_streamlines.py -h
    scil_bundle_volume_per_label.py -h
    scil_bundle_mean_std.py -h
    scil_json_merge_entries.py -h

    touch ${prefix}__length_stats.json
    touch ${prefix}__endpoints_map_raw.json
    touch ${prefix}__endpoints_metric_stats.json
    touch ${prefix}__mean_std.json
    touch ${prefix}__volume.json
    touch ${prefix}__streamline_count.json
    touch ${prefix}__volume_per_label.json
    touch ${prefix}__mean_std_per_point.json
    touch ${prefix}_endpoints_map_head.nii.gz
    touch ${prefix}_endpoints_map_tail.nii.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: 2.0.2
    END_VERSIONS
    """
}
