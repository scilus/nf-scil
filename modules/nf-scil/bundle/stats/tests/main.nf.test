nextflow_process {

    name "Test Process BUNDLE_STATS"
    script "../main.nf"
    process "BUNDLE_STATS"

    tag "modules"
    tag "modules_nfcore"
    tag "bundle"
    tag "bundle/stats"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    setup {
        run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
            script "../../../../../subworkflows/nf-scil/load_test_data/main.nf"
            process {
                """
                input[0] = Channel.from( [ "tractometry.zip" ] )
                input[1] = "test.load-test-data"
                """
            }
        }
    }

    test("bundle - stats") {
        config "./nextflow.config"
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/IFGWM.trk"),
                        file("\${test_data_directory}/IFGWM_labels_map.nii.gz"),
                        file("\${test_data_directory}/mni_masked.nii.gz"),
                        []
                    ]}
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("bundle - stats - light") {
        config "./nextflow_light.config"
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/IFGWM{,_color}.trk"),
                        file("\${test_data_directory}/IFGWM{_labels_map,}.nii.gz"),
                        file("\${test_data_directory}/mni_masked.nii.gz"),
                        []
                    ]}
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("bundle - stats - lesions") {
        config "./nextflow_lesions.config"
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/IFGWM.trk",),
                        file("\${test_data_directory}/IFGWM_labels_map.nii.gz"),
                        file("\${test_data_directory}/mni_masked.nii.gz"),
                        file("\${test_data_directory}/IFGWM.nii.gz")
                    ]}
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
