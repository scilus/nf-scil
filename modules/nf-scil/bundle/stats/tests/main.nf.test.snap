{
    "bundle - stats - lesions": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_length_stats.json:md5,72873da12072e254f369ec316911df28"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_endpoints_map_raw.json:md5,2a86a4d10878c942c09723c3122940b4"
                    ]
                ],
                "10": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_mean_std_per_point.json:md5,efcac000fb6c0f607b7377154e68854c"
                    ]
                ],
                "11": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__lesion_stats.json:md5,cefff165378a416816c1c7a644c4969c"
                    ]
                ],
                "12": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "IFGWM_endpoints_map_head.nii.gz:md5,444b8e3e40f4694b84a1fb5a2be8a52f"
                    ]
                ],
                "13": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "IFGWM_endpoints_map_tail.nii.gz:md5,a4cb0339f71c17a8703e920e0eb4916a"
                    ]
                ],
                "14": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__IFGWM.trk_lesion_map.nii.gz:md5,390c8abfcfa2533367099c9fac26ee21"
                    ]
                ],
                "15": [
                    "versions.yml:md5,c22283763660c5778cdd9c34c64200bd"
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_endpoints_metric_stats.json:md5,d13e29a5d9dd09e8b5a2d40b3b61b3a9"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_mean_std.json:md5,f7463e5519063f7e0f32709d0dfdc1a2"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_volume.json:md5,f2cf5bbf678b0d364f26120ad10900ae"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_volume_lesions.json:md5,6221f9954d93904de992d5b7af99bcce"
                    ]
                ],
                "6": [
                    
                ],
                "7": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_streamline_count_lesions.json:md5,08433569dd80f6b1dd53193e7cf446d6"
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_volume_per_label.json:md5,1ba8e9d51cc2dec86fdaa4498488424b"
                    ]
                ],
                "9": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_volume_per_label_lesions.json:md5,b9138245af55e3becd3c9e2bec2c9821"
                    ]
                ],
                "endpoints_head": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "IFGWM_endpoints_map_head.nii.gz:md5,444b8e3e40f4694b84a1fb5a2be8a52f"
                    ]
                ],
                "endpoints_metric_stats": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_endpoints_metric_stats.json:md5,d13e29a5d9dd09e8b5a2d40b3b61b3a9"
                    ]
                ],
                "endpoints_raw": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_endpoints_map_raw.json:md5,2a86a4d10878c942c09723c3122940b4"
                    ]
                ],
                "endpoints_tail": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "IFGWM_endpoints_map_tail.nii.gz:md5,a4cb0339f71c17a8703e920e0eb4916a"
                    ]
                ],
                "length": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_length_stats.json:md5,72873da12072e254f369ec316911df28"
                    ]
                ],
                "lesion_map": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__IFGWM.trk_lesion_map.nii.gz:md5,390c8abfcfa2533367099c9fac26ee21"
                    ]
                ],
                "lesion_stats": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__lesion_stats.json:md5,cefff165378a416816c1c7a644c4969c"
                    ]
                ],
                "mean_std": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_mean_std.json:md5,f7463e5519063f7e0f32709d0dfdc1a2"
                    ]
                ],
                "mean_std_per_point": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_mean_std_per_point.json:md5,efcac000fb6c0f607b7377154e68854c"
                    ]
                ],
                "streamline_count": [
                    
                ],
                "streamline_count_lesions": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_streamline_count_lesions.json:md5,08433569dd80f6b1dd53193e7cf446d6"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c22283763660c5778cdd9c34c64200bd"
                ],
                "volume": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_volume.json:md5,f2cf5bbf678b0d364f26120ad10900ae"
                    ]
                ],
                "volume_lesions": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_volume_lesions.json:md5,6221f9954d93904de992d5b7af99bcce"
                    ]
                ],
                "volume_per_labels": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_volume_per_label.json:md5,1ba8e9d51cc2dec86fdaa4498488424b"
                    ]
                ],
                "volume_per_labels_lesions": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_volume_per_label_lesions.json:md5,b9138245af55e3becd3c9e2bec2c9821"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0-rc1",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-01T13:47:29.33467"
    },
    "bundle - stats": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_length_stats.json:md5,72873da12072e254f369ec316911df28"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_endpoints_map_raw.json:md5,2a86a4d10878c942c09723c3122940b4"
                    ]
                ],
                "10": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_mean_std_per_point.json:md5,efcac000fb6c0f607b7377154e68854c"
                    ]
                ],
                "11": [
                    
                ],
                "12": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "IFGWM_endpoints_map_head.nii.gz:md5,444b8e3e40f4694b84a1fb5a2be8a52f"
                    ]
                ],
                "13": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "IFGWM_endpoints_map_tail.nii.gz:md5,a4cb0339f71c17a8703e920e0eb4916a"
                    ]
                ],
                "14": [
                    
                ],
                "15": [
                    "versions.yml:md5,c22283763660c5778cdd9c34c64200bd"
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_endpoints_metric_stats.json:md5,d13e29a5d9dd09e8b5a2d40b3b61b3a9"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_mean_std.json:md5,f7463e5519063f7e0f32709d0dfdc1a2"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_volume.json:md5,f2cf5bbf678b0d364f26120ad10900ae"
                    ]
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "8": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_volume_per_label.json:md5,1ba8e9d51cc2dec86fdaa4498488424b"
                    ]
                ],
                "9": [
                    
                ],
                "endpoints_head": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "IFGWM_endpoints_map_head.nii.gz:md5,444b8e3e40f4694b84a1fb5a2be8a52f"
                    ]
                ],
                "endpoints_metric_stats": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_endpoints_metric_stats.json:md5,d13e29a5d9dd09e8b5a2d40b3b61b3a9"
                    ]
                ],
                "endpoints_raw": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_endpoints_map_raw.json:md5,2a86a4d10878c942c09723c3122940b4"
                    ]
                ],
                "endpoints_tail": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "IFGWM_endpoints_map_tail.nii.gz:md5,a4cb0339f71c17a8703e920e0eb4916a"
                    ]
                ],
                "length": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_length_stats.json:md5,72873da12072e254f369ec316911df28"
                    ]
                ],
                "lesion_map": [
                    
                ],
                "lesion_stats": [
                    
                ],
                "mean_std": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_mean_std.json:md5,f7463e5519063f7e0f32709d0dfdc1a2"
                    ]
                ],
                "mean_std_per_point": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_mean_std_per_point.json:md5,efcac000fb6c0f607b7377154e68854c"
                    ]
                ],
                "streamline_count": [
                    
                ],
                "streamline_count_lesions": [
                    
                ],
                "versions": [
                    "versions.yml:md5,c22283763660c5778cdd9c34c64200bd"
                ],
                "volume": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_volume.json:md5,f2cf5bbf678b0d364f26120ad10900ae"
                    ]
                ],
                "volume_lesions": [
                    
                ],
                "volume_per_labels": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_volume_per_label.json:md5,1ba8e9d51cc2dec86fdaa4498488424b"
                    ]
                ],
                "volume_per_labels_lesions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0-rc1",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-01T13:42:36.266056"
    },
    "bundle - stats - light": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "10": [
                    
                ],
                "11": [
                    
                ],
                "12": [
                    
                ],
                "13": [
                    
                ],
                "14": [
                    
                ],
                "15": [
                    "versions.yml:md5,c22283763660c5778cdd9c34c64200bd"
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_streamline_count.json:md5,bedf1fbd223dd47d9a8970dc779f296f"
                    ]
                ],
                "7": [
                    
                ],
                "8": [
                    
                ],
                "9": [
                    
                ],
                "endpoints_head": [
                    
                ],
                "endpoints_metric_stats": [
                    
                ],
                "endpoints_raw": [
                    
                ],
                "endpoints_tail": [
                    
                ],
                "length": [
                    
                ],
                "lesion_map": [
                    
                ],
                "lesion_stats": [
                    
                ],
                "mean_std": [
                    
                ],
                "mean_std_per_point": [
                    
                ],
                "streamline_count": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_streamline_count.json:md5,bedf1fbd223dd47d9a8970dc779f296f"
                    ]
                ],
                "streamline_count_lesions": [
                    
                ],
                "versions": [
                    "versions.yml:md5,c22283763660c5778cdd9c34c64200bd"
                ],
                "volume": [
                    
                ],
                "volume_lesions": [
                    
                ],
                "volume_per_labels": [
                    
                ],
                "volume_per_labels_lesions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0-rc1",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-01T13:42:46.785058"
    }
}